src/Wallabag/ApiBundle/Controller/WallabagRestController.php

```diff
@@ -43,16 +43,16 @@ public function getEntriesAction(Request $request)
     {
         $this->validateAuthentication();
 
-        $isArchived = (int) $request->query->get('archive');
-        $isStarred = (int) $request->query->get('starred');
+        $isArchived = (null === $request->query->get('archive')) ? null : (bool) (int) $request->query->get('archive');
+        $isStarred = (null === $request->query->get('starred')) ? null : (bool) (int) $request->query->get('starred');
```

Comment:
When someone send a string instead of an integer, I wanted to cast it in integer before. But I can change it if you prefer (your code is more readable).